api/search.go

```diff
@@ -1,28 +1,644 @@
 package api
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
 	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
 
 	"github.com/TUM-Dev/gocast/dao"
+	"github.com/TUM-Dev/gocast/model"
 	"github.com/TUM-Dev/gocast/tools"
 	"github.com/gin-gonic/gin"
+	"github.com/meilisearch/meilisearch-go"
 )
 
-func configGinSearchRouter(router *gin.Engine, daoWrapper dao.DaoWrapper) {
-	routes := searchRoutes{daoWrapper}
+func configGinSearchRouter(router *gin.Engine, daoWrapper dao.DaoWrapper, meiliSearchInstance tools.MeiliSearchInterface) {
+	routes := searchRoutes{daoWrapper, meiliSearchInstance}
 
 	searchGroup := router.Group("/api/search")
+	searchGroup.GET("", routes.search)
 	withStream := searchGroup.Group("/stream/:streamID")
 	withStream.Use(tools.InitStream(daoWrapper))
 	withStream.GET("/subtitles", routes.searchSubtitles)
+
+	searchGroup.GET("/courses", routes.searchCourses)
 }
 
 type searchRoutes struct {
 	dao.DaoWrapper
+	m tools.MeiliSearchInterface
 }
 
+const (
+	FilterMaxCoursesCount = 3
+	DefaultLimit          = 10
+)
+
+type MeiliSearchMap map[string]any
+
 func (r searchRoutes) searchSubtitles(c *gin.Context) {
 	s := c.MustGet("TUMLiveContext").(tools.TUMLiveContext).Stream
 	q := c.Query("q")
-	c.JSON(http.StatusOK, tools.SearchSubtitles(q, s.ID))
+	c.JSON(http.StatusOK, r.m.SearchSubtitles(q, s.ID))
+}
+
+/*
```

Comment:
nit: This comment seems sort of random and out of place.
- it's written in german and
- does not follow the style of the other doc-comments

=> is this intentional?